File: src/main/java/de/byjoker/myjfql/command/CommandService.java
package de.byjoker.myjfql.command;

import java.util.List;

public interface CommandService {

    void registerCommand(Command command);

    void unregisterCommand(Command command);

    void searchCommands(String path);

    boolean existsCommand(String name);

    void execute(CommandSender sender, String name);

    Command getCommand(String name);

    List<Command> getCommands();

}

File: src/main/java/de/byjoker/myjfql/command/CommandService.kt
package de.byjoker.myjfql.command

import de.byjoker.myjfql.core.MyJFQL
import de.byjoker.myjfql.lang.Interpreter

interface CommandService {

    fun registerCommand(command: Command)
    fun unregisterCommand(name: String)
    fun getCommand(name: String): Command?
    fun getCommands(): List<Command>
    fun execute(sender: CommandSender, query: String, interpreter: Interpreter = MyJFQL.getInstance().interpreter)

}
